### PR TITLE
Thinking through long term location model

### PR DESCRIPTION
**Pull request by daveshah**
Given the constrains with the Google places API, as well as some of the functionality we may want down the road, I'm putting this branch out there to help think through some things.

### CODE REVIEW

File: priv/repo/migrations/20160907190212_create_location.exs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  def change do
+    create table(:locations) do
+      add :postal_code, :string
+      add :state, :string
+      add :city, :string
+      add :latitude, :float
+      add :longitude, :float
+
+      timestamps()
```
Comment (mbramson) [inline priv/repo/migrations/20160907190212_create_location.exs]: Do we want a street address as well? I could see this being useful for both runs and clubs.

Comment (mbramson) [inline priv/repo/migrations/20160907190212_create_location.exs]: Do we want locations to have a name?

File: web/models/location.ex [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  schema "locations" do
+    field :postal_code, :string
+    field :state, :string
+    field :city, :string
+    field :latitude, :float
+    field :longitude, :float
+
+    timestamps()
+  end
+
```
Comment (mbramson) [inline web/models/location.ex]: Should we use a variable here to store the required and optional fields like:
```
  @required_fields [:postal_code, :state, :city, :latitude, :longitude]
  @optional_fields []
```

File: web/models/location.ex [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  @doc """
+  Builds a changeset based on the `struct` and `params`.
+  """
+  def changeset(struct, params \\ %{}) do
+    struct
+    |> cast(params, [:postal_code, :state, :city, :latitude, :longitude])
+    |> validate_required([:postal_code, :state, :city, :latitude, :longitude])
```
Comment (mbramson) [inline web/models/location.ex]: That way we could change this to 
```
|> cast(params, @required_fields, @optional_fields)
|> validate_required(@required_fields)
```
and maybe make it a bit cleaner?

File: web/models/location.ex (hunk 2 above)
Comment (daveshah) [inline web/models/location.ex]: Yep - I'm a fan of this.

File: priv/repo/migrations/20160907190212_create_location.exs (hunk 1 above)
Comment (daveshah) [inline priv/repo/migrations/20160907190212_create_location.exs]: I don't think for this model - this all came from seed data (```priv/repo/locations.csv```). The thing I was thinking about was using this as a cross reference. This needs a lot of thinking through from a high level with use cases I have in mind. I should probably enumerate those use cases as that would help :)

**Pull request closed without merging**